openvpn/session/credentials/provider.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package credentials
+
+import (
+	"github.com/mysterium/node/session"
+	ovpnsession "github.com/mysterium/node/openvpn/session"
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/openvpn/middlewares/client/auth"
+)
+
+// SignatureCredentialsProvider returns session id as username and id signed with given signer as password
+func SignatureCredentialsProvider(id session.SessionID, signer identity.Signer) auth.CredentialsProvider {
+	return func() (string, string, error) {
```

Comment:
Returns list is hard to understand, especially with 2 unnamed strings. You have to just to `auth.CredentialsProvider` to understand what this function returns.

Naming return values solves this:
```
	return func() (username string, password string, err error) {
		signature, err := signer.Sign([]byte(ovpnsession.SignaturePrefix + id))
		return string(id), signature.Base64(), err
	}
```